npm/pkg/dataplane/policies/policymanager_linux.go

```diff
@@ -63,18 +180,43 @@ func (pMgr *PolicyManager) addPolicy(networkPolicy *NPMNetworkPolicy, _ map[stri
 }
 
 func (pMgr *PolicyManager) removePolicy(networkPolicy *NPMNetworkPolicy, _ map[string]string) error {
-	chainsToDelete := chainNames([]*NPMNetworkPolicy{networkPolicy})
+	if !pMgr.IPTablesInBackground {
+		return pMgr.removeAllPolicies([]*NPMNetworkPolicy{networkPolicy})
+	}
+
+	hasIngress, hasEgress := networkPolicy.hasIngressAndEgress()
+	d := Ingress
+	if hasIngress && hasEgress {
+		d = Both
+	} else if hasEgress {
+		d = Egress
+	}
+
+	oi := &opInfo{
+		op:          remove,
+		direction:   d,
+		wasInKernel: networkPolicy.inLinuxKernel,
+	}
+
+	pMgr.policyMap.linuxDirtyCache[networkPolicy.PolicyKey] = append(pMgr.policyMap.linuxDirtyCache[networkPolicy.PolicyKey], oi)
+	return nil
+}
+
+func (pMgr *PolicyManager) removeAllPolicies(networkPolicies []*NPMNetworkPolicy) error {
+	chainsToDelete := chainNames(networkPolicies)
 	creator := pMgr.creatorForRemovingPolicies(chainsToDelete)
 
 	// Stop reconciling so we don't contend for iptables, and so we don't update the staleChains at the same time as reconcile()
 	pMgr.reconcileManager.forceLock()
 	defer pMgr.reconcileManager.forceUnlock()
 
-	// 1. Delete jump rules from ingress/egress chains to ingress/egress policy chains.
-	// We ought to delete these jump rules here in the foreground since if we add an NP back after deleting, iptables-restore --noflush can add duplicate jump rules.
-	deleteErr := pMgr.deleteOldJumpRulesOnRemove(networkPolicy)
-	if deleteErr != nil {
-		return npmerrors.SimpleErrorWrapper("failed to delete jumps to policy chains", deleteErr)
+	for _, networkPolicy := range networkPolicies {
+		// 1. Delete jump rules from ingress/egress chains to ingress/egress policy chains.
+		// We ought to delete these jump rules here in the foreground since if we add an NP back after deleting, iptables-restore --noflush can add duplicate jump rules.
+		deleteErr := pMgr.deleteOldJumpRulesOnRemove(networkPolicy)
+		if deleteErr != nil {
```

Comment:
need to ignore errors here

Reply:
we actually already ignore `doesNotExistErrorCode` in case the rule doesn't exist